src/v/bytes/random.cc

```diff
@@ -53,7 +56,9 @@ iobuf split_as_iobuf(std::string_view str, int n_fragments) {
     iobuf res;
     auto cur = str.begin();
     for (auto fragment_len : fragment_lengths) {
-        res.append(ss::temporary_buffer<char>(cur, fragment_len));
+        // force add a new fragment
```

Comment:
`split_as_iobuf` should be documented that the original data must still be held to as long as iobuf is alive? Or just copy the string to avoid putting additional requirements on the caller. Should be ok for testing code?

Reply:
`fragmented_iobuf`?